File: backend/services/card_service.py
from common import json, random, Path
class CardService:
    def __init__(self):
        self.data_path = Path(__file__).parent.parent / "data"
        self.genders = self._load_genders()
    def _load_genders(self):
        try:
            with open(self.data_path / "genders.json", encoding="utf-8") as f:
                return json.load(f)
        except FileNotFoundError:
            print("Erro: genders.json não encontrado")
            return []
        except json.JSONDecodeError as e:
            print(f"Erro ao decodificar JSON: {e}")
            return []
    def get_random_item(self, array):
        return random.choice(array)
    def create_bullying_card(self, id, male_text, female_text):
        return {
            "id": id,
            "type": "twoOptions",
            "aggressorGender": self.get_random_item(self.genders),
            "male": {
                "text": male_text
            },
            "female": {
                "text": female_text
            },
            "options": {
                "firstOption": {
                    "text": "Não fazer nada"
                },
                "secondOption": {
                    "mode": "select",
                    "text": {
                        "male": "Bater nele",
                        "female": "Bater nela"
                    },
                    "options": {
                        "male": [
                            {"label": "Chute na virilha", "value": "groinKick"},
                            {"label": "Soco no queixo", "value": "chinPunch"}
                        ],
                        "female": [
                            {"label": "Puxar o cabelo", "value": "hairPull"},
                            {"label": "Empurrar para longe", "value": "push"}
                        ]
                    }
                }
            }
        }

File: backend/services/cardService.js
import fs from "fs";
import path from "path";
import { getRandomItem } from "./utilities.js";

class CardService {
  constructor() {
    this.dataPath = path.join(__dirname, "..", "data");
    this.genders = this._loadGenders();
  }
  _loadGenders() {
    try {
      const filePath = path.join(this.dataPath, "genders.json");
      return JSON.parse(fs.readFileSync(filePath, "utf-8"));
    } catch (error) {
      if (error.code === "ENOENT") {
        console.error("Erro: genders.json não encontrado");
      } else {
        console.error("Erro ao decodificar JSON:", error);
      }
      return [];
    }
  }

  createBullyingCard(id, maleText, femaleText) {
    return {
      id,
      type: "twoOptions",
      aggressorGender: getRandomItem(this.genders),
      male: {
        text: maleText,
      },
      female: {
        text: femaleText,
      },
      options: {
        firstOption: {
          text: "Não fazer nada",
        },
        secondOption: {
          mode: "select",
          text: {
            male: "Bater nele",
            female: "Bater nela",
          },
          options: {
            male: [
              { label: "Chute na virilha", value: "groinKick" },
              { label: "Soco no queixo", value: "chinPunch" },
            ],
            female: [
              { label: "Puxar o cabelo", value: "hairPull" },
              { label: "Empurrar para longe", value: "push" },
            ],
          },
        },
      },
    };
  }
}

export default CardService;
